developer/src/kmcmpdll/Compiler.cpp

```
@@ -3679,70 +3739,87 @@ BOOL IsValidCallStore(PFILE_STORE fs)
 
-  buf = new BYTE[len + 1]; // null terminated
-  outbuf = new WCHAR[len + 1];
-  if (ReadFile(hInfile, buf, len, &len2, NULL)) {
+  buf = new KMX_BYTE[len + 1];  // null terminated
+  outbuf = new KMX_WCHAR[len + 1];
+
+  len2= fread(buf,1,len,fp_in);
+  if (len2) {
     buf[len2] = 0;
     p = buf;
     poutbuf = outbuf;
     if (hasPreamble) {
       // We have a preamble, so we attempt to read as UTF-8 and allow conversion errors to be filtered. This is not great for a
       // compiler but matches existing behaviour -- in future versions we may not do lenient conversion.
       ConversionResult cr = ConvertUTF8toUTF16(&p, &buf[len2], (UTF16 **)&poutbuf, (const UTF16 *)&outbuf[len], lenientConversion);
-      WriteFile(hOutfile, outbuf, (DWORD)(INT_PTR)(poutbuf - outbuf) * 2, &len2, NULL);
+      fwrite(outbuf, (KMX_DWORD)(INT_PTR)(poutbuf - outbuf) * 2 , 1, fp_out);
     }
     else {
       // No preamble, so we attempt to read as strict UTF-8 and fall back to ANSI if that fails
       ConversionResult cr = ConvertUTF8toUTF16(&p, &buf[len2], (UTF16 **)&poutbuf, (const UTF16 *)&outbuf[len], strictConversion);
       if (cr == sourceIllegal) {
-        // Not a valid UTF-8 file, so fall back to ANSI
-        AddCompileMessage(CHINT_NonUnicodeFile);
-        poutbuf = strtowstr((PSTR)buf);
-        WriteFile(hOutfile, poutbuf, (DWORD)wcslen(poutbuf) * 2, &len2, NULL);
-        delete[] poutbuf;
+        // Not a valid UTF-8 file, so fall back to ANSI      
+        // AddCompileMessage(CHINT_NonUnicodeFile);
+        // note, while this message is defined, for now we will not emit it
+        // because we don't support HINT/INFO messages yet and we don't want
+        // this to cause a blocking compile at this stage
+        // do strtowstr only when no invalid characters are found
+        if( p==0){
+          poutbuf = strtowstr((PKMX_STR)buf);
+          fwrite(poutbuf, (KMX_DWORD)u16len(poutbuf) * 2 , 1, fp_out);
+          delete[] poutbuf;
+        }
```

Don't we want to output some message if `p != 0` instead of doing nothing and silently swallowing the error?

---

👍